java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -92,11 +85,30 @@ private CallOption[] getOptions() {
    * @param flightInfo The {@link FlightInfo} instance from which to fetch results.
    * @return a {@code FlightStream} of results.
    */
-  public List<FlightStream> getStreams(final FlightInfo flightInfo) {
-    return flightInfo.getEndpoints().stream()
-        .map(FlightEndpoint::getTicket)
-        .map(ticket -> sqlClient.getStream(ticket, getOptions()))
-        .collect(Collectors.toList());
+  public List<FlightStream> getStreams(final FlightInfo flightInfo) throws RuntimeException {
+    try {
+      ArrayList<FlightStream> streams = new ArrayList<>();
+      for (FlightEndpoint ep : flightInfo.getEndpoints()) {
+        for (Location loc : ep.getLocations()) {
```

Comment:
We are not supposed to iterate over locations. For a given endpoint/ticket, multiple locations can be returned for redundancy and load balancing reasons and the caller can arbitrarily choose one.

It's also legal for no location to be returned, in which case the caller should just use the same node that getFlightInfo was called on. Note that if at least one location is returned, the caller cannot assume that getStream() can be called on the same node as getFlightInfo().

Reply:
Thanks for this info! I'll put together a more serious PR and undraft it, taking this into account.